From 8ca5737b301aa2fe62dcb749c9c559195b534dd3 Mon Sep 17 00:00:00 2001
From: Marcel Apfelbaum <marcel.a@redhat.com>
Date: Wed, 6 Nov 2013 16:32:35 +0100
Subject: [PATCH 76/81] gdbstub: fix for commit 87f25c12bfeaaa0c41fb857713bbc7e8a9b757dc

RH-Author: Marcel Apfelbaum <marcel.a@redhat.com>
Message-id: <1383755557-21590-8-git-send-email-marcel.a@redhat.com>
Patchwork-id: 55552
O-Subject: [RHEL-7 qemu-kvm PATCH v3 7/9] gdbstub: fix for commit 87f25c12bfeaaa0c41fb857713bbc7e8a9b757dc
Bugzilla: 990601
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

This commit used the wrong check to prevent an assertion failure.
After this commit, you need to start a guest in the monitor, you
cannot use anymore the "c" command in the debugger.  This is
undesirable.  The commit's aim was to prevent a restart
after a KVM internal error or something like that; use
runstate_needs_reset() for that.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Reviewed-by: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: 1370272015-9659-2-git-send-email-pbonzini@redhat.com
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit 26ac7a31fbf5522d2ca3f0e2e5b5c8e915701f66)
Signed-off-by: Marcel Apfelbaum <marcel.a@redhat.com>
---
 gdbstub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 gdbstub.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gdbstub.c b/gdbstub.c
index 90e54cb..e8541f3 100644
--- a/gdbstub.c
+++ b/gdbstub.c
@@ -371,7 +371,7 @@ static inline void gdb_continue(GDBState *s)
 #ifdef CONFIG_USER_ONLY
     s->running_state = 1;
 #else
-    if (runstate_check(RUN_STATE_DEBUG)) {
+    if (!runstate_needs_reset()) {
         vm_start();
     }
 #endif
-- 
1.7.1

